velox/exec/PartitionedOutputBufferManager.cpp

```diff
@@ -373,11 +372,16 @@ bool PartitionedOutputBuffer::deleteResults(int destination) {
       return false;
     }
     freed = buffer->deleteResults();
+    dataAvailable = buffer->getAndClearNotify();
```

Comment:
Could we consider to pass DataAvailable& to deleteResults() so make it in one method instead of two? Otherwise we might forget to get DataAvailable for new use case in the future. Thanks!

Reply:
This is a good point. This is an existing pattern in this class though:

```
          buffer->enqueue(sharedData);
          dataAvailableCallbacks.emplace_back(buffer->getAndClearNotify());
```

It would also be nice to switch to using folly::Futures. We can make these changes in follow-up PRs.